File: crates/breez-sdk/cli/src/command/stable_balance.rs
use breez_sdk_spark::{BreezSdk, StableBalanceActiveLabel, UpdateUserSettingsRequest};
use clap::Subcommand;

use crate::command::print_value;

#[derive(Clone, Debug, Subcommand)]
pub enum StableBalanceCommand {
    /// Get the stable balance active label
    Get,
    /// Set the stable balance active label
    Set {
        /// The label to activate (e.g. "USDB")
        label: String,
    },
    /// Unset stable balance
    Unset,
}

pub async fn handle_command(
    sdk: &BreezSdk,
    command: StableBalanceCommand,
) -> Result<bool, anyhow::Error> {
    match command {
        StableBalanceCommand::Get => {
            let settings = sdk.get_user_settings().await?;
            print_value(&settings.stable_balance_active_label)?;
            Ok(true)
        }
        StableBalanceCommand::Set { label } => {
            sdk.update_user_settings(UpdateUserSettingsRequest {
                spark_private_mode_enabled: None,
                stable_balance_active_label: Some(StableBalanceActiveLabel::Set { label }),
            })
            .await?;
            let settings = sdk.get_user_settings().await?;
            print_value(&settings)?;
            Ok(true)
        }
        StableBalanceCommand::Unset => {
            sdk.update_user_settings(UpdateUserSettingsRequest {
                spark_private_mode_enabled: None,
                stable_balance_active_label: Some(StableBalanceActiveLabel::Unset),
            })
            .await?;
            let settings = sdk.get_user_settings().await?;
            print_value(&settings)?;
            Ok(true)
        }
    }
}

File: crates/breez-sdk/bindings/examples/cli/langs/wasm/src/stable-balance.js
'use strict'

const { Command } = require('commander')
const { printValue } = require('./serialization')

/**
 * Register all stable-balance subcommands on the given commander program.
 *
 * @param {Command} program - The parent commander program (or subcommand)
 * @param {() => object} getSdk - Function that returns the SDK instance
 */
function registerStableBalanceCommands(program, getSdk) {
  const stableBalance = program
    .command('stable-balance')
    .description('Stable balance related commands')

  // --- get ---
  stableBalance
    .command('get')
    .description('Get the stable balance active label')
    .action(async () => {
      const sdk = getSdk()
      const settings = await sdk.getUserSettings()
      printValue(settings.stableBalanceActiveLabel)
    })

  // --- set ---
  stableBalance
    .command('set')
    .description('Set the stable balance active label')
    .argument('<label>', 'The label to activate (e.g. "USDB")')
    .action(async (label) => {
      const sdk = getSdk()
      await sdk.updateUserSettings({
        sparkPrivateModeEnabled: undefined,
        stableBalanceActiveLabel: { type: 'set', label }
      })
      const settings = await sdk.getUserSettings()
      printValue(settings)
    })

  // --- unset ---
  stableBalance
    .command('unset')
    .description('Unset stable balance')
    .action(async () => {
      const sdk = getSdk()
      await sdk.updateUserSettings({
        sparkPrivateModeEnabled: undefined,
        stableBalanceActiveLabel: { type: 'unset' }
      })
      const settings = await sdk.getUserSettings()
      printValue(settings)
    })
}

module.exports = { registerStableBalanceCommands }
